uamqp/_transport.py

```diff
@@ -524,7 +524,7 @@ def _read(self, toread, initial=False, buffer=None,
         try:
             while toread:
                 try:
-                    nbytes = self.sock.recv_into(view[nbytes:])
+                    nbytes = self.sock.recv_into(view[length:])
```

Comment:
hmm this doesn't break anything?

Reply:
nope, it doesn't break -- I would categorize it as a bug fix.

with small frames, nbytes == length, so we're good.

however, with large frames which are split into several reads, we have to move forward the cursor ( which is the `length` ) to properly append the read bytes into the memory view. otherwise we're just inserting the read bytes at the position of `nbytes` which will override bytes in the memoryview.